src/arcaflow_plugin_sdk/test_schema.py

```diff
@@ -603,58 +734,109 @@ def test_unserialize_embedded(self):
         )
         self.assertEqual(unserialized_data.a, "Hello world!")
 
-        unserialized_data2: Basic3 = s.unserialize(
-            {discriminator_field_name: "b", "b": 42}
+        unserialized_data2: DoubleInlineStr = s.unserialize(
+            {
+                discriminator_field_name: "a2",
+                default_discriminator: "a",
+                "msg": "Hi again",
+            }
         )
-        self.assertIsInstance(unserialized_data2, Basic3)
-        self.assertEqual(unserialized_data2.b, 42)
+        self.assertIsInstance(unserialized_data2, DoubleInlineStr)
+        self.assertEqual(unserialized_data2.msg, "Hi again")
 
     def test_validation(self):
-        s = schema.OneOfStringType[InlineStr](
+        s = schema.OneOfStringType[Basic](
             {
-                "a": schema.RefType("a", self.scope_mixed_type),
-                "b": schema.RefType("b", self.scope_mixed_type),
+                "a": schema.RefType("a", self.scope_basic),
+                "b": schema.RefType("b", self.scope_basic),
             },
-            scope=self.scope_mixed_type,
+            scope=self.scope_basic,
             discriminator_field_name=discriminator_field_name,
+            discriminator_inlined=False,
         )
-
         with self.assertRaises(ConstraintException):
             # noinspection PyTypeChecker
             s.validate(InlineStr(None, "Hello world!"))
-
         with self.assertRaises(ConstraintException):
+            # noinspection PyTypeChecker
             s.validate(InlineStr("b", "Hello world!"))
+        s.validate(Basic("Hello world!"))
```

Comment:
We do not need a discriminated value because our argument includes an instance of a class in the union, `Basic`,  (object with id `a`, and discriminated value `a` is associated with the class `Basic` through its schema).